src/Core/src/Extensions/IViewExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Microsoft.Maui.Extensions
+{
+	// TODO: Make this public on .NET 8
+	internal static class IViewExtensions
+	{
+		internal static bool GetIsEnabled(this IView view)
+		{
+			bool parentIsEnabled = true;
+
+			if (view.Parent is IView parentView)
+			{
+				parentIsEnabled = parentView.GetIsEnabled();
+			}
+
+			return parentIsEnabled && view.IsEnabled;
```

Comment:
If view.IsEnabled is false, then none of the rest of this method matters. That should be checked first, before checking ancestors.

Reply:
That is right, updated.